tools/mitohifi/mitohifi.xml

```diff
@@ -0,0 +1,349 @@
+<tool id='mitohifi' name='MitoHiFi' version='@TOOL_VERSION@+@GALAXY_TOOL_VERSION@' profile='20.01'>
+    <description>assembly mitogenomes from Pacbio HiFi reads</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro='edam_ontology' />
+    <expand macro='requirements' />
+    <version_command>python mitohifi_v2.py -h</version_command>
+    <command detect_errors='exit_code'><![CDATA[
+        #if $operation_mode.command == 'find_reference'
+            findMitoReference.py 
+            --species '${operation_mode.species}'
+            --email '${operation_mode.email}'
+            --outfolder ./ 
+            #if $operation_mode.min_length
+                --min_length $operation_mode.min_length
+            #end if
+            $operation_mode.exact_specie
+            && mv ./*fasta ./reference_genome.fasta
+            && mv ./*gb ./reference_genome.gb
+        #elif $operation_mode.command == 'mitohifi'
+            mitohifi_v2.py 
+            #if $operation_mode.input_option.input == 'pacbio'
+                -r '${ ' '.join(["%s" % $x for $x in $operation_mode.input_option.input_reads]) }'
+            #else
+                -c '${operation_mode.input_option.input_assembly}'
+            #end if
+            -f '${operation_mode.reference_fasta}'
+            -g '${operation_mode.reference_genbank}'
+            #if $operation_mode.advanced_options.query_blast
+                -p $operation_mode.advanced_options.query_blast
+            #end if
+            #if $operation_mode.advanced_options.circular_size
+                --circular-size $operation_mode.advanced_options.circular_size
+            #end if
+            #if $operation_mode.advanced_options.circular_offset
+                --circular_offset $operation_mode.advanced_options.circular_offset
+            #end if
+            -o $operation_mode.genetic_code
+            -t \${GALAXY_SLOTS:-1}
+        #end if
+        && ls -lah
+    ]]>    </command>
+    <inputs>
+        <conditional name="operation_mode">
+            <param name="command" type="select" label="Operation type selector" help="Select a type of operation">
+                <option value="find_reference">Find a close-related mitochondrial reference genome</option>
+                <option value="mitohifi">Run MitoHiFi</option>
+            </param>
+            <when value="find_reference">
+                <param argument="--species" type="text" value="" label="Specie name" help="Specie name in binomial nomenclature">
+                    <sanitizer invalid_char="">
```

Comment:
@gallardoalba general question, those should be validators not sanitizers, is that correct? https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-validator

Reply:
I try to restrict the use of the wrong parameters which could be dangerous (hacking?).  Then I think that I should use sanitize, but not sure about it.